### ISSUE TITLE
manual_saturating_arithmetic for unsigned `checked_sub().unwrap_or_default()`

### ISSUE THREAD

**Issue by skunne** [rust-lang/rust-clippy#17365]
### Summary

Lint `manual_saturating_arithmetic` correctly suggests I use `saturating_sub` if I write one of these two:
```rust
fn some_math(a: usize, b: usize) {
    let x = a.checked_sub(b).unwrap_or(usize::MIN);
    let y = a.checked_sub(b).unwrap_or(0);
}
```
But the lint stays silent for this one:
```rust
fn some_math(a: usize, b: usize) {
    let z = a.checked_sub(b).unwrap_or_default();
}
```
I'm much more likely to use `.unwrap_or_default()` than `.unwrap_or(usize::MIN)`.

### Lint Name

manual_saturating_arithmetic

### Reproducer

I tried this code:

```rust
fn some_math(a: usize, b: usize) {
    let z = a.checked_sub(b).unwrap_or_default();
}
```

I expected to see this happen:
```
  --> src/main.rs
 |     let z = a.checked_sub(b).unwrap_or_default();
 |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: consider using `saturating_sub`: `a.saturating_sub(b)`

```
Instead, this happened:
```
Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.06s
```

### Version

```text
rustc 1.92.0 (ded5c06cf 2025-12-08)
binary: rustc
commit-hash: ded5c06cf21d2b93bffd5d884aa6e96934ee4234
commit-date: 2025-12-08
host: x86_64-unknown-linux-gnu
release: 1.92.0
LLVM version: 21.1.3
```

**Comment by Gri-ffin** [comment]
This correctly lints on the latest stable. Fixed by #15845
